draft-ietf-ppm-dap.md

```diff
@@ -1401,34 +1409,52 @@ prep_state = VDAF.prep_init(vdaf_verify_key,
                             report_id,
                             public_share,
                             plaintext_input_share.payload)
-out = VDAF.prep_next(prep_state, None)
+(prep_state, prep_share) = VDAF.prep_next(prep_state, None)
 ~~~
 
 `vdaf_verify_key` is the VDAF verification key shared by the aggregators;
-`agg_id` is the aggregator ID (`0x00` for the Leader and `0x01` for the helper);
+`agg_id` is the aggregator ID (`0x00` for the Leader and `0x01` for the Helper);
 `agg_param` is the opaque aggregation parameter distributed to the aggregators
 by the collector; `public_share` is the public share generated by the client and
 distributed to each aggregator; and `plaintext_input_share` is the aggregator's
 PlaintextInputShare.
 
-If either step fails, the aggregator marks the report as invalid with error
+At this point, the Leader would send the resulting preparation share in its
+`AggregationJobInitReq` to the Helper in its
+`ReportAggregationInitReq.leader_prep_share` field, and continue this process
+once it receives a response from the Helper. The Helper would continue this
+process immediately.
+
+Next, aggregators run the following procedure to receive the next step:
```

Comment:
Only the Helper runs prep_shares_to_prep.

Previously, the Leader had gathered the shares, combined them into the message, then sends the message to the Helper; now the Helper gathers the shares, combines them into the message, and sends the message to the Leader.

Reply:
Argh, that's what I meant to write. I split this out explicitly now.

One thought: the `input-share-prep` section now describes two slightly-different procedures for the Leader & Helper. WDYT about moving this section into the `leader-init` & `helper-init` sections? I think this would likely be clearer, especially now that the aggregators are following slightly different processes, and would match the structure of the continuation section.